### DISCUSSION TITLE
can we use easy effect in voip programs such as google meet,discord,Teams etc....

### DISCUSSION THREAD

**Discussion by 7549brijesh** [wwmm/easyeffects#1036] [General]
I am searching for the AI powered noise suppression in linux there are very little and working like rnnoise,NoiseTorch,cadmus When I used Noisetorch it say my device is (?incompatible) I am finding cadmus a little complex now I am having a very little hope with easy effects

If there is any way to use easy effect in voip programs please let me know

and I am using easy effect in google meet I get this:
![2021-07-19_17-19](https://user-images.githubusercontent.com/25589680/126156103-01a4be7d-7a9a-48d5-8e99-f39ff627a930.png)

using two different device can cause echo

I am doing something worng?

**Comment by wwmm** [comment]
I would not bother with that warning. I use EasyEffects with Microsoft Teams and our virtual devices do not cause echo by themselves. If you put your mic in front of your speakers you will have echo even if using your sound card device directly.

**Comment by 7549brijesh** [comment]
can you guide me how to setup easy effect for google meet,discord,Team I want to use it as noise supression software (Just like krisp software in windows) I am using Boart Rockerz 510 which has a inbuilt mic in it

**Reply by wwmm** [reply]
I am not an expert on noise suppression. For now I only add our plugin based on rnnoise to the microphone pipeline
![Screenshot from 2021-07-20 10-44-16](https://user-images.githubusercontent.com/16225172/126334731-6b026e3d-5f54-4c65-99fe-9539c51a2776.png)
Although rnnoise is a filter based on a machine learning algorithm that is trained for specific situations it has been surprisingly good in all of my microphones. It takes all of noises away from the mic signal as long as you have only voice and some background noise.

Echo that is a little more complicated to handle. The echo canceller I wrote based on the speex library still is a little unreliable.

**Reply by wwmm** [reply]
There was one occasion where one of my students microphone had noises and adding the Noise Reduction plugin to the output effects pipeline helped me to listen only his voice. But usually I use it only in the mic pipeline.

**Reply by 7549brijesh** [reply]
Thanks a lots guys I was waiting for this answer
